File: client/src/app/configurator/models/DungeonConfig.ts
import {NPC} from "./NPCConfig";
import {Item} from "./Item";
import {RaceConfig} from "./RaceConfig";
import {ClassConfig} from "./ClassConfig";
import {CommandConfig} from "./CommandConfig";
import {RoomConfigExport} from "../components/configuration/configuration.component";

export interface DungeonConfig {
  name: string;
  startRoom: string;
  startEquipment: Item[];
  rooms: StringMap<RoomConfigExport>;
  npcConfigs: StringMap<NPC>;
  itemConfigs: StringMap<Item>;
  races: RaceConfig[];
  classes: ClassConfig[];
  commandConfig: CommandConfig;
}


export type StringMap<V> = {
  [key: string]: V
}

File: server/src/main/kotlin/net/bmgames/state/DungeonConfig.kt
package net.bmgames.state

import kotlinx.serialization.Serializable
import net.bmgames.state.model.*


/**
 * DTO for configurations of a game
 * Mirrors [Game] but omits some fields
 * */
@Serializable
data class DungeonConfig(
    val name: String,
    val races: List<Race>,
    val classes: List<Clazz>,
    val commandConfig: CommandConfig,
    val npcConfigs: Map<String, NPC>,
    val itemConfigs: Map<String, Item>,
    val startEquipment: List<Item>,
    val startRoom: String,
    val rooms: Map<String, Room>
)
